cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -362,20 +365,21 @@ public ApplicationDetail getLatestAppDetail(ApplicationReference appRef) throws
    * that don't exist.
    * @throws Exception if failed to get details.
    */
-  public Map<ApplicationId, ApplicationDetail> getAppDetails(Collection<ApplicationId> appIds) throws Exception {
+  public Map<ApplicationId, ApplicationDetail> getAppDetails(Collection<ApplicationId> appIds,
+                                                             boolean getSourceControlMeta) throws Exception {
     Set<? extends EntityId> visibleIds = accessEnforcer.isVisible(new HashSet<>(appIds),
                                                                   authenticationContext.getPrincipal());
     Set<ApplicationId> filterIds = appIds.stream().filter(visibleIds::contains).collect(Collectors.toSet());
     Map<ApplicationId, ApplicationMeta> appMetas = store.getApplications(filterIds);
     Map<ApplicationId, String> principals = ownerAdmin.getOwnerPrincipals(filterIds);
-
-    return appMetas.entrySet().stream()
-             .collect(Collectors.toMap(
-               Entry::getKey,
-               entry -> ApplicationDetail.fromSpec(entry.getValue().getSpec(),
-                                                   principals.get(entry.getKey()),
-                                                   entry.getValue().getChange())
-             ));
+    
+    return appMetas.entrySet().stream().collect(Collectors.toMap(
+      Entry::getKey,
+      entry -> ApplicationDetail.fromSpec(entry.getValue().getSpec(),
+                                          principals.get(entry.getKey()),
+                                          entry.getValue().getChange(),
+                                          getSourceControlMeta ? entry.getValue().getSourceControlMeta() : null)
```

Comment:
Why is source control meta pulled separately? Can it be made as part of applicationDetail like every other metadata?

Reply:
Because when we write app configs into git, we don't want to include the `SourceControlMeta`, which currently is the `fileHash`. This information is also useless since it's like chicken/egg question: we could only get the fileHash from git after we complete the write and commit, but the real fileHash also depends on the one we saved from CDF DB, these two are always different